### PR TITLE
Push packaging work into the upstream repo

### PR DESCRIPTION
**Pull request by sdodson**
Squashed commit of the following:

commit bcbe40d4df7c30af5159ab112df268e70525075e
Author: dobbymoodge <jolamb@redhat.com>
Date:   Wed Feb 11 15:11:48 2015 -0500

    Add "Requires=" to SDN service files

    To ensure startup of `openshift-sdn-*` services, a `Requires=` line has
    been added that will induce the related service dependencies to also
    start.

    Bug: 1189422
    https://bugzilla.redhat.com/show_bug.cgi?id=1189422

commit 4afe9e17dbb2b465867488de2fb3183967d89110
Author: Scott Dodson <sdodson@redhat.com>
Date:   Wed Feb 11 14:31:00 2015 -0500

    etcd is http rather than https

commit 1efa3b42f33b1933cb7982af2a93d52c6b5ae5d4
Author: dobbymoodge <jolamb@redhat.com>
Date:   Fri Jan 30 17:24:41 2015 -0500

    Automatic commit of package [openshift-sdn] release [0.4-1].

commit 504e38c55c4c3f836acec41a47b9a0f5494e1ce2
Author: dobbymoodge <jolamb@redhat.com>
Date:   Fri Jan 30 17:21:52 2015 -0500

    fix Source0 line in specfile

commit 94034c1ad6cde8efd094f04e07924e4e739a413d
Author: dobbymoodge <jolamb@redhat.com>
Date:   Fri Jan 30 17:07:26 2015 -0500

    Automatic commit of package [openshift-sdn] release [0.3-1].

commit 296c19deefea20f8ccf6d2427fa55760d6e9810d
Author: dobbymoodge <jolamb@redhat.com>
Date:   Fri Jan 30 14:16:51 2015 -0500

    Add DOCKER_OPTIONS to sysconfig files

commit 02b4ab39338c7fad1a9244ce02e0b7661c35bf01
Author: Scott Dodson <sdodson@redhat.com>
Date:   Fri Jan 30 11:02:10 2015 -0500

    Automatic commit of package [openshift-sdn] release [0.2-1].

commit 365e63b1f00ed153e674de73512caaa80236a315
Author: Scott Dodson <sdodson@redhat.com>
Date:   Fri Jan 30 10:18:41 2015 -0500

    Anything that may be expanded to multiple argv should not be enclosed in braces

commit 1a8b0a81cc3e49d23c28d4e034c001b4da05943d
Author: Scott Dodson <sdodson@redhat.com>
Date:   Thu Jan 29 16:14:55 2015 -0500

    EnvironmentFile values should not be quoted

commit edc3032b89e0fdc67756564bbdb5d6a1fa07e533
Author: Scott Dodson <sdodson@redhat.com>
Date:   Thu Jan 29 13:28:58 2015 -0500

    Enclose environment variables in curly braces

    This is necessary whenever the variable isn't on its own

commit 7b0b80232d06960677dff35d3ea7297153dbf837
Author: Scott Dodson <sdodson@redhat.com>
Date:   Thu Jan 29 11:18:38 2015 -0500

    Fixup requires

    Make openshift-sdn-master and openshift-sdn-node require openshift-sdn of
    matching versions.

    Get rid of version requirement on openshift for now. This may change in the
    future.

commit 5d511a157a24eec2f52b6b2a472cc8c847b07e30
Author: dobbymoodge <jolamb@redhat.com>
Date:   Thu Jan 29 11:17:52 2015 -0500

    Remove erroneous openshift dependency

    Also snipped out some debugging output

commit a77c166fd8ed07da9f984de79d954bd645339019
Author: dobbymoodge <jolamb@redhat.com>
Date:   Wed Jan 28 16:55:03 2015 -0500

    Master/Node subpackages, systemd unit files

    Created `openshift-sdn-master` and `openshift-sdn-node` roles via
    subpackages and targeted `systemd` configs

commit 81adaa349ad2562f961663b4325a0ae05d37c087
Author: dobbymoodge <jolamb@redhat.com>
Date:   Wed Jan 28 11:11:31 2015 -0500

    Added missing Requires:

commit a1e301e21aaeaf23783c54cb101e447df620d362
Author: dobbymoodge <jolamb@redhat.com>
Date:   Tue Jan 27 15:37:44 2015 -0500

    openshift-sdn-simple-setup-node.sh installed but unpackaged

    Added `openshift-sdn-simple-setup-node.sh` to `%{files}`

commit b58d07e48855eb7aae32995bfb53a07d416525c9
Author: dobbymoodge <jolamb@redhat.com>
Date:   Tue Jan 27 15:26:08 2015 -0500

    Minor fixups - Source0 URL, Version, install

    `Source0` referenced the wrong repository, `Version` makes more sense as
    `0.1` right now, `%setup` macro didn't get committed last time,
    `openshift-sdn-simple-setup-node.sh` needs to be `install`ed from `$PWD`

commit ccd7e7d38e4d72266c0f862b0a484ca56ec2e3f6
Author: dobbymoodge <jolamb@redhat.com>
Date:   Tue Jan 27 15:11:16 2015 -0500

    fixed up %setup line

commit 50b458073ac7f0a6a7bec2ffc36ccb24c773ac1f
Author: dobbymoodge <jolamb@redhat.com>
Date:   Mon Jan 26 15:31:15 2015 -0500

    Initialized to use tito, added specfile

    Added openshift-sdn.spec file, releasers.conf

### CODE REVIEW

**Comment by sdodson** [comment]
@dobbymoodge just wanted to get this pushed upstream as there have been a few issues filed upstream against the packaging.

**Comment by sdodson** [comment]
Resolves #16

File: rel-eng/openshift-sdn-master.sysconfig [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# options set to the defaults documented above.
+#
+# Example:
+#   OPTIONS="-etcd-keyfile=/path/to/keyfile"
+OPTIONS=
+
+# The $DOCKER_OPTIONS variable is used to overwrite the $OPTIONS
+# variable in the /etc/sysconfig/docker file. This is a pretty ugly
+# hack, but whatevs
+DOCKER_OPTIONS='-b=lbr0 --mtu=1450 --selinux-enabled'
```
Comment (rajatchopra) [inline rel-eng/openshift-sdn-master.sysconfig]: We do not need to set any docker options for the master, do we?
We probably do not even require docker as a package for the master.

Comment (mrunalp) [inline rel-eng/openshift-sdn-master.sysconfig]: Yeah, master should have no dependency on docker.

Comment (sdodson) [inline rel-eng/openshift-sdn-master.sysconfig]: ACK, removed DOCKER_OPTIONS from master sysconfig.

File: openshift-sdn.spec [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+Requires:       bridge-utils
+
+%description
+%{summary}
+
+%package master
+Summary:        Openshift SDN Master
+Requires:       openshift-sdn = %{version}-%{release}
+Requires:       openshift-master
+
```
Comment (nak3) [inline openshift-sdn.spec]: Need these Requires for master and node
~~~
Requires(post): systemd
Requires(preun): systemd
Requires(postun): systemd
~~~
http://fedoraproject.org/wiki/Packaging:ScriptletSnippets#New_Packages

File: openshift-sdn.spec [hunk 3]
```diff
@@ -0,0 +1,141 @@
+#debuginfo not supported with Go
+%global debug_package %{nil}
+%global gopath      %{_datadir}/gocode
+%global osdn_gopath _output/local/go
+%global gopkg_base  github.com/openshift
+%global import_path %{gopkg_base}/openshift-sdn
+%global commit      1d03a6c75a44a638b9f2590b1475a6c980ce829d
```
Comment (nak3) [inline openshift-sdn.spec]: no such a commit "1d03a6c75a44a638b9f2590b1475a6c980ce829d" in openshift-sdn git log.
It means conflict with tar download with link written in Source0

File: openshift-sdn.spec [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+Name:           openshift-sdn
+Version:        0.4
```
Comment (nak3) [inline openshift-sdn.spec]: version 0.1 ?

File: openshift-sdn.spec [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+Summary:        Openshift SDN Node
+Requires:       openshift-sdn = %{version}-%{release}
+Requires:       openshift-node
+Requires:       docker-io >= 1.3.2
+
+%description node
+%{summary}
+
+%prep
+%setup -q
```
Comment (nak3) [inline openshift-sdn.spec]: When extract tar of source0 we will find openshift-sdn-%{commit},
so should be 
`%setup -q -n openshift-sdn-%{commit}`

File: openshift-sdn.spec (hunk 4 above)
Comment (nak3) [inline openshift-sdn.spec]: Sorry, it looks like 0.4 is correct. but it doesn't have git tag version.

**Comment by sdodson** [comment]
@nak3 Ok, everything you've mentioned should be addressed now.

**Comment by nak3** [comment]
@sdodson  Thank you! LGTM

**Comment by sdodson** [comment]
@rajatchopra I think this should be good to merge now. To your point, the docker package is only required by the openshift-sdn-node subpackage.

**Comment by dobbymoodge** [comment]
+1

**Comment by rajatchopra** [comment]
@sdodson Thanks for productizing it.
At some point, we might want to asunder any indication that this software runs with only OpenShift. Looks good today, though.

**Pull request merged**